### DISCUSSION TITLE
How to add a new block type?

### DISCUSSION THREAD

**Discussion by MonstraG** [outline/rich-markdown-editor#486] [Q&A] [answered]
I'm interested in ability to insert voice recordings, with something like a `/record` or `/voice`, I was trying to copy and modify implementation of Image page block, but I didn't really know if I'm doing the right thing.

So, is there a guide or a reference to add a new block type?

**Comment by tommoor** [comment, marked as answer]
Hi, the nodes map directly to Markdown – as there is no audio tag in markdown there isn't really a way to achieve this reliably while maintaining the aim of the project.

**Reply by MonstraG** [reply]
Yep, that's why I'm not requesting to add this audio feature to this project, but to provide a (hopefully, simple) way for me to add new types of blocks in my application only.

**Reply by tommoor** [reply]
The best bet might be to hijack something like the [notice](https://github.com/outline/rich-markdown-editor/blob/main/src/nodes/Notice.tsx) node. The markdown could then be something like:

```
:::audio
http://url…
:::
```

Regardless of whether the change is merged the content needs to be serializable to markdown

**Reply by MonstraG** [reply]
Alright, I'll try that, thanks for the tip)
